code/_onclick/other_mobs.dm

```diff
@@ -41,6 +41,14 @@
 /mob/living/carbon/RestrainedClickOn(atom/A)
 	return 0
 
+//Ranged override abilities for carbons
+/mob/living/ClickOn(atom/A, params)
+	if(ranged_ability && !stat)
+		ranged_ability.use_ability(A, src, params) //Spit at something, use a fireball, etc.

```

Comment:
You just define ```/THING/proc/InterceptClickOn(clicker, params, clicked_thing)```
It also is client based, but use your imagination, add a var to mobs that allow them to pass these intercepts to clients during ```Login()```